File: C++/DynamicProgramming/LongestPalindromicSubsequence.cpp
#include <bits/stdc++.h>
#include "LongestCommonSubsequence.cpp"
using namespace std;

int longestPalindromicSubsequence(string str)
{
  string rev = str;
  reverse(rev.begin(), rev.end());
  return longestCommonSubsequenceTabulation(str, rev);
}

// int main()
// {
//   cout << longestPalindromicSubsequence("saippuakivikauppias") << endl;
// }

File: Java/dynamicprogramming/LongestPalindromicSubsequence.java
package dynamicprogramming;

public class LongestPalindromicSubsequence {
  public int longestPalindromicSubsequence(String str) {
    LongestCommonSubsequence ob = new LongestCommonSubsequence();
    return ob.longestCommonSubsequenceTabulation(str, new StringBuilder(str).reverse().toString());
  }

  public static void main(String[] args) {
    LongestPalindromicSubsequence ob = new LongestPalindromicSubsequence();
    System.out.println(ob.longestPalindromicSubsequence("saippuakivikauppias"));
  }
}
